app/src/org/odk/collect/android/jr/extensions/PollSensorAction.java

```diff
@@ -110,13 +117,19 @@ public void run() {
      * @param providers Set of String objects that may contain LocationManager.GPS_PROVDER and/or LocationManager.NETWORK_PROVIDER
      */
     private void requestLocationUpdates(Set<String> providers) {
-        if (providers.isEmpty()) {
+        Context context = CommCareApplication._().getApplicationContext();
+        if (providers.isEmpty() &&
+                (ContextCompat.checkSelfPermission(context, Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED ||
+                        ContextCompat.checkSelfPermission(context, Manifest.permission.ACCESS_COARSE_LOCATION) == PackageManager.PERMISSION_GRANTED)) {
             mLocationManager.removeUpdates(PollSensorAction.this);
             return;
         }
         
         for (String provider : providers) {
-            mLocationManager.requestLocationUpdates(provider, 0, 0, PollSensorAction.this);
+            if ((provider.equals(LocationManager.GPS_PROVIDER) && ContextCompat.checkSelfPermission(context, Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED) ||
+                    (provider.equals(LocationManager.NETWORK_PROVIDER) && ContextCompat.checkSelfPermission(context, Manifest.permission.ACCESS_COARSE_LOCATION) == PackageManager.PERMISSION_GRANTED)) {
+                mLocationManager.requestLocationUpdates(provider, 0, 0, PollSensorAction.this);
```

Comment:
Does this mean we're just skipping the use of the background provider unless someone already accepted GPS permissions instead of asking users to do them?